internal/download/download.go

```diff
@@ -71,3 +81,93 @@ func DownloadFile(url, outPath, header, token string) error {
 
 	return nil
 }
+
+// DownloadRegistryOpenAPIBundle Returns a file path within the downloaded bundle or error
+func DownloadRegistryOpenAPIBundle(ctx context.Context, namespaceID, reference, outPath string) (string, error) {
+	serverURL := auth.GetServerURL()
+	insecurePublish := false
+	if strings.HasPrefix(serverURL, "http://") {
+		insecurePublish = true
+	}
+	reg := strings.TrimPrefix(serverURL, "http://")
+	reg = strings.TrimPrefix(reg, "https://")
+
+	bundleLoader := loader.NewLoader(loader.OCILoaderOptions{
+		Registry: reg,
+		Access: ocicommon.NewRepositoryAccess(config.GetSpeakeasyAPIKey(), namespaceID, ocicommon.RepositoryAccessOptions{
+			Insecure: insecurePublish,
+		}),
+	})
+
+	bundleResult, err := bundleLoader.LoadOpenAPIBundle(ctx, reference)
+	if err != nil {
+		return "", err
+	}
+
+	buf, err := io.ReadAll(bundleResult.Body)
+	if err != nil {
+		return "", err
+	}
+	defer bundleResult.Body.Close()
+
+	reader := bytes.NewReader(buf)
+	zipReader, err := zip.NewReader(reader, int64(len(buf)))
+	if err != nil {
+		return "", err
+	}
+
+	outputFileName := ""
+	for _, file := range zipReader.File {
+		if strings.Contains(file.Name, ".yaml") { // TODO: this is not going to fly
```

Comment:
This is the main thing that needs to change. I did this just so I could make sure it works with all test cases right now. But I recognize this can't be permanent.

What I actually need is to grab the root openapi file that the workflow should reference from within this bundle. @disintegrator any good ideas on how to consistently do that.

Reply:
I think we need to add our own custom annotation on the manifest that contains the location of the root document in the bundle blob.

Reply:
The bundler does know what the root document is though?